chartfx-chart/src/test/java/de/gsi/chart/renderer/spi/ContourDataSetRendererTests.java

```diff
@@ -0,0 +1,121 @@
+import org.testfx.framework.junit5.ApplicationExtension;
+import org.testfx.framework.junit5.Start;
+
+import de.gsi.chart.XYChart;
+import de.gsi.chart.axes.spi.DefaultNumericAxis;
+import de.gsi.chart.renderer.ContourType;
+import de.gsi.chart.ui.utils.JavaFXInterceptorUtils.SelectiveJavaFxInterceptor;
+import de.gsi.chart.utils.FXUtils;
+import de.gsi.dataset.DataSet3D;
+
+/**
+ * Tests {@link de.gsi.chart.renderer.spi.ContourDataSetRenderer }
+ *
+ * @author rstein
+ *
+ */
+@ExtendWith(ApplicationExtension.class)
+@ExtendWith(SelectiveJavaFxInterceptor.class)
+public class ContourDataSetRendererTests {
+    private static final Class<?> clazz = ContourDataSetRendererTests.class;
+    private static final Logger LOGGER = LoggerFactory.getLogger(clazz);
+    private static final String className = clazz.getSimpleName();
+    private static final String referenceFileName = "./Reference_" + className;
+    private static final String referenceFileExtension = ".png";
+    private static final int MAX_TIMEOUT_MILLIS = 1000;
+    private static final int WAIT_N_FX_PULSES = 3;
+    private static final double IMAGE_CMP_THRESHOLD = 0.99; // 1.0 is perfect identity
+    private static final int WIDTH = 300;
+    private static final int HEIGHT = 200;
+    private XYChart chart;
+    private ContourDataSetRenderer renderer;
+    private Image testImage;
+
+    @Start
+    public void start(Stage stage) {
+        assertDoesNotThrow(() -> new ContourDataSetRenderer());
+        renderer = new ContourDataSetRenderer();
+        renderer.getDatasets().add(getTestDataSet());
+        chart = new XYChart(new DefaultNumericAxis(), new DefaultNumericAxis());
+        chart.getRenderers().set(0, renderer);
+        chart.legendVisibleProperty().set(true);
+
+        stage.setScene(new Scene(chart, WIDTH, HEIGHT));
+        stage.show();
+    }
+
+    @ParameterizedTest
+    @EnumSource(ContourType.class)
+    public void testRendererNominal(final ContourType contourType) throws IOException, Exception {
+        testRenderer(contourType, false);
+    }
+
+    @ParameterizedTest
+    @EnumSource(ContourType.class)
+    public void testRendererAlt(final ContourType contourType) throws IOException, Exception {
+        testRenderer(contourType, true);
+    }
+
+    private void testRenderer(final ContourType contourType, final boolean altImplementation) throws IOException, Exception {
+        renderer.setAltImplementation(altImplementation);
+        renderer.setContourType(contourType);
+        final String contourTypeString = renderer.getContourType().toString();
+        final String referenceImage = referenceFileName + contourTypeString + (altImplementation ? "_ALT" : "") + referenceFileExtension;
+        FXUtils.runAndWait(() -> chart.requestLayout());
+        assertTrue(FXUtils.waitForFxTicks(chart.getScene(), WAIT_N_FX_PULSES, MAX_TIMEOUT_MILLIS));
+
+        FXUtils.runAndWait(() -> testImage = chart.snapshot(null, null));
```

Comment:
This is an explicit check that the 'wait for the FX thread' has finished properly and hasn't time-out and/or was aborted prematurely. This is not necessary for every interaction with the FX thread only for those where the pulse response (layout etc.) is critical and/or need to be checked afterwards. Also, this does not work meaningfully if called from the FX thread itself. Thus adding it to e.g. the `@TestFx` environment makes IMHO little sense (also because not every FX interaction affects the pulse).